File: app/services/api/src/dtos/input/room.py
from typing import List, Optional

from pydantic import BaseModel, Field


class CreateRoom(BaseModel):
    """
    DTO for creating a room
    """
    institution_id: str
    name: str
    capacity: int = Field(default=30, ge=1)
    features: List[str] = Field(default_factory=list)


class UpdateRoom(BaseModel):
    """
    DTO for updating a room
    """
    name: Optional[str] = None
    capacity: Optional[int] = None
    features: Optional[List[str]] = None

File: app/services/ui/src/types/room.ts
export interface RoomData {
  id?: string;
  _id?: string;
  institution_id: string;
  name: string;
  capacity: number;
  features?: string[];
}

export interface RoomShape {
  id: string;
  institution_id: string;
  name: string;
  capacity: number;
  features: string[];
}

export class Room implements RoomShape {
  id: string;
  institution_id: string;
  name: string;
  capacity: number;
  features: string[];

  constructor(data: RoomData) {
    this.id = data.id ?? data._id ?? '';
    this.institution_id = data.institution_id;
    this.name = data.name;
    this.capacity = data.capacity;
    this.features = data.features ?? [];
  }

  static from(data: RoomData): Room {
    return new Room(data);
  }
}

